File: RedSheet.DbEntities/Models/ProjectStakeholder.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ProjectStakeholders", Schema = "dbo")]
	[Serializable]
	public partial class ProjectStakeholder  
	{
 
	[Required] 
	[MaxLength(250)]
	public string Frequancy { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectStakeholderId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string StakeholderName { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 
 
	[RelationshipTableAttribue("StakeholderTypes","dbo")]
	//Relationships
	public int StakeholderTypeId { get; set; } 
 
	[ForeignKey("StakeholderTypeId")]
	public StakeholderType StakeholderType { get; set; } 
	[InverseProperty("ProjectStakeholder")]
	public ICollection<StakeholderCommunicationMode> StakeholderCommunicationModes { get; set; } 


	public ProjectStakeholder()
	{
	this.StakeholderCommunicationModes = new HashSet<StakeholderCommunicationMode>();

    }

    }
}

File: redsheet-portal/src/app/database-models/project-stakeholder.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule, StakeholderType, StakeholderCommunicationMode, vProjectStakeholderRecord } from './'
export class ProjectStakeholder {
    constructor(projectStakeholder?: ProjectStakeholder | vProjectStakeholderRecord) {
        let properties = ["frequancy", "projectStakeholderId", "stakeholderName", "projectModuleId", "stakeholderTypeId", "stakeholderCommunicationModes",];
        for (let property of properties)
            if (projectStakeholder && projectStakeholder[property])
                this[property] = projectStakeholder[property];
    }

    @required()
    @maxLength(200)
    frequancy: string = undefined;

    projectStakeholderId: number = 0;

    @required()
    @maxLength(100)
    stakeholderName: string = undefined;

    @range(0, 2147483647)
    projectModuleId: number = undefined;
    projectModule: ProjectModule;

    @range(0, 2147483647)
    stakeholderTypeId: number = undefined;
    stakeholderType: StakeholderType;
    @nested(StakeholderCommunicationMode)
    stakeholderCommunicationModes: StakeholderCommunicationMode[];



}
